### ISSUE TITLE
Arcane Missiles incorrectly interact with Arcane Blast debuff stacks.

### ISSUE THREAD

**Issue by Riptidemyth** [SunwellTracker/issues#1520]
Decription: 
Arcane Missiles incorrectly interact with Arcane Blast debuff stacks.

How it works: 
As it stands, the arcane blast stacks are removed only when Arcane Missiles cast is finished. That leads to the problem: if one casts Missiles (2.5 sec cast) when Blast debuff duration left is <2 sec, what happens is first Missiles' projectiles are buffed with 4x Blast debuff, but few last ones (that are shoot when debuff is gone) are not.

How it should work:
All stacks of Arcane Blast debuff should be consumed at the START of the Missiles cast and all Missiles' projectiles should be buffed accordingly.

Source (you should point out proofs of your report, please give us some source):
Regardless of the timespan, it seems to work this way throughout wotlk. Here are some clips from that time:
https://youtu.be/VLUMN1Gy2mE?t=23 (Ulduar)
https://youtu.be/tlxcNMltpU8?t=98 (ICC)

**Comment by gabbat** [comment]
correct. AB stacks should get consumed as soon as Arcane missiles (with or without related procC) or Arcane barrage is cast.
